src/Event/Preprocess/AbstractPreprocessEvent.php

```diff
@@ -41,6 +51,23 @@ public static function name() {
     return static::DISPATCH_NAME_PREFIX . static::getHook();
   }
 
+  /**
+   * Get the Event dispatcher type.
+   *
+   * When no event dispatcher type is provided it means that
+   * event is known instead of based on a suggestion so we
+   * return the static name().
+   *
+   * @return string
+   *   Event dispatcher type.
+   */
+  public function getDispatcherType() {
```

Comment:
This breaks with the current preprocess classes and why do the if check in this method. When your sending out suggestions you know you are doing this so a specific method is more appropriate.